File: java/src/merge_sort/randomStringTestData.java
package merge_sort;

import java.util.Arrays;
import java.util.Comparator;
import java.util.Random;

class randomStringTestData {
	private static final Random rand = new Random(0);

	public String[] array, expectedArray;

	randomStringTestData() {
		int n = rand.nextInt(50), max = rand.nextInt(1_000_000_000);
		array = new String[n];
		for (int i = 0; i < n; i++) {
			array[i] = String.valueOf(rand.nextInt(1 + max));
		}
		expectedArray = array.clone();
		Arrays.sort(expectedArray, Comparator.comparingInt(String::length));
	}
}

File: go/merge_sort/random_string_test_data.go
package merge_sort

import (
	"fmt"
	"sort"
)

type randomStringTestData struct {
	array, expectedArray []string
}

func newRandomStringTestData() *randomStringTestData {
	var td randomStringTestData

	n, maxVal := random.IntN(50), random.IntN(1_000_000_000)
	td.array = make([]string, n)
	for i := range n {
		td.array[i] = fmt.Sprint(random.IntN(1 + maxVal))
	}
	td.expectedArray = append([]string{}, td.array...)
	sort.Slice(td.expectedArray, func(i, j int) bool {
		return len(td.expectedArray[i]) < len(td.expectedArray[j])
	})

	return &td
}
